tools/datafix/refresh_ids.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+""" Utility to update the datastore key of each Bug to the new format determined by the pre put hook.
+
+    Does this by deleting and reputting each Bug entry.
+"""
+from google.cloud import ndb
+import osv
+
+import argparse
+import json
+import functools
+import time
+import typing
+
+MAX_BATCH_SIZE = 500
+
+
+def get_relevant_ids(verbose: bool) -> list[str]:
+  relevant_ids = []
+
+  query = osv.Bug.query()
+  query.projection = ["db_id"]
+  print(f"Running initial query on {query.kind}...")
+
+  result: typing.Iterable[osv.Bug] = query.iter()
+  counter = 0
+
+  for res in result:
+    counter += 1
+    # Check if the key needs to be updated
+    if res.key.id() != res.db_id:  # type: ignore
+      relevant_ids.append(res.db_id)
+      if verbose:
+        print(res.db_id + ' - ' + res.key.id())  # type: ignore
+
+  print(f"Found {len(relevant_ids)} / {counter} relevant bugs to refresh.")
+  return relevant_ids
+
+
+def reput_bugs(dryrun: bool, verbose: bool) -> None:
+  """ Reput all bugs from a given source."""
+
+  # Uncomment below to load the state and skip the get_relevant_ids func
+  # relevant_ids = json.load(open('relevant_ids.json', 'r'))
+  relevant_ids = get_relevant_ids(verbose)
+
+  # Store the state incase we cancel halfway to avoid having to do the initial query again
+  json.dump(relevant_ids, open('relevant_ids.json', 'w'))
```

Comment:
nit: the file doesn't get (explicitly) closed.

Reply:
Fixed